File: src/java/SimulationObjects/SimulationObject.java
package SimulationObjects;

import java.io.Serializable;

/*
 * created by: b farooq, poly montreal
 * on: 22 october, 2013
 * last edited by: b farooq, poly montreal
 * on: 22 october, 2013
 * summary: 
 * comments:
 */

public class SimulationObject implements Serializable
{
    protected int myID;

    public int GetID()
    {
        return myID;
    }

    public void SetID(int id)
    {
        myID = id;
    }

    protected int myType;
    public int GetAgentType()
    {
        return myType;
    }

    public void SetAgentType(int curTyp)
    {
        myType = curTyp;
    }

    public String GetNewJointKey(String baseDim)
    {
        return "";
    }

    public SimulationObject CreateNewCopy(String baseDim, int baseDimVal)
    {
        return null;
    }
}

File: src/c#/SimulationObjects/SimulationObject.cs
/*
 * created by: b farooq, poly montreal
 * on: 22 october, 2013
 * last edited by: b farooq, poly montreal
 * on: 22 october, 2013
 * summary: 
 * comments:
 */

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SimulationObjects
{
    [Serializable]
    class SimulationObject
    {
        protected AgentType Type;

        public AgentType GetAgentType()
        {
            return Type;
        }
        public void SetAgentType(AgentType curTyp)
        {
            Type = curTyp;
        }
        public virtual string GetNewJointKey(string baseDim)
        {
            return "";
        }
        public virtual SimulationObject CreateNewCopy(string baseDim,
            int baseDimVal)
        {
            return null;
        }
    }
}
